src/Node/ScenarioInterface.php

```diff
@@ -17,4 +17,12 @@
  */
 interface ScenarioInterface extends ScenarioLikeInterface, TaggedNodeInterface
 {
+    /**
+     * @todo Altering an interface feels like a BC break. What's the alternative?
```

Comment:
What should we do here?

I think the following approach would also work and should theoretically be backword compatible:
```php
// instead of this:

    static fn (ScenarioInterface $scenario) => $scenario
        ->withSteps(...

// using this:

    static fn (ExampleNode|ScenarioNode|OutlineNode $scenario) => $scenario
        ->withSteps(...
```

Reply:
Indeed - adding a method to an interface is definitely a BC break, but I agree it's not required.